inst/pages/multimedia.qmd

```diff
@@ -0,0 +1,590 @@
+library(multimedia)
+library(stringr)
+library(ggplot2)
+
+
+##################
+# Load iHMP data #
+##################
+
+# Load data
+se_relative <- curatedMetagenomicData(
+  "HMP_2019_ibdmdb.relative_abundance", 
+  dryrun = FALSE
+  )[[1]]
+
+# Normalize to proportions (from %)
+assay(se_relative) <- assay(se_relative) / 100
+
+# Change the rownames names for variables of interest
+rownames(se_relative) <- sub('.*s__', '', rownames(se_relative))
+
+
+########################
+# Reference nonIBD set #
+########################
+
+# Reference set: healthy
+ref_set <- colData(se_relative)$disease == "healthy"
+
+
+########################################
+# Calculate Bray-Curtis dissimilarity #
+########################################
+
+# Bray-Curtis dissimilarity
+dist <- as.matrix(vegdist(t(assay(se_relative)), method = "bray", na.rm = TRUE))
+
+# Assign SampleID for matching
+colData(se_relative)$SampleID <- colnames(se_relative)
+
+
+#################################
+# Calculate the dysbiosis score #
+#################################
+
+# Calculate dysbiosis score for each sample
+colData(se_relative)$dysbiosis <- sapply(seq_along(ref_set), function(i) {
+  median(dist[i, ref_set & 
+              (colData(se_relative)$SampleID != colData(se_relative)$SampleID[i])], 
+         na.rm = TRUE)
+})
+
+
+#############################################################
+# Subset to IBD only and only one post-baseline time point #
+#############################################################
+
+# Keep IBD only
+se_relative <- se_relative[, colData(se_relative)$disease != "healthy"]
+
+# Visit 1 (baseline) or 25 (post)
+se_relative <- se_relative[, colData(se_relative)$visit_number %in% c(1, 21)]
+
+# Keep subjects with both visits
+keep_subjects <- names(which(table(colData(se_relative)$subject_id) == 2))
+se_relative <- se_relative[, colData(se_relative)$subject_id %in% keep_subjects]
+
+# Define time point label
+colData(se_relative)$Time_point <- ifelse(
+  colData(se_relative)$visit_number == 1, "T0", "T1"
+)
+
+```
+
+Next, we will define the mediation anlaysis components. By using the processed 
+data, we define:
+
+-   Treatment: time point (baseline vs. post-baseline)
+-   Outcome: dysbiosis score
+-   Mediators: scaled species abundances
+
+These are bundled into a mediation data object (exper), which is passed into the 
+multimedia function to estimate path models for the treatment-to-mediator 
+(α path), mediator-to-outcome (β path), and the combined indirect effect (α×β 
+path).
+
+```{r define-mediation-data-frame}
+
+#############################
+# Define mediation data set #
+#############################
+
+# Treatment: Pre and Post (Baseline vs. Post-baseline)
+treatment <- factor(colData(se_relative)$Time_point, levels = c("T0", "T1"))
+
+# Outcome: Dysbiosis Score
+dysbiosis <- as.numeric(colData(se_relative)$dysbiosis)
+
+# Mediators (scaled, directly from SE): Species
+M <- t(assay(se_relative))
+M <- scale(M)
```

Comment:
Use mia functions and keep the new assay together with the data object:


```
tse <- transformAssay(tse, assay.type="relabundance", method="standardize", name="scaled")
```

Then you can always call:

```
assay(tse, "scaled")
```

It is intended that you can keep multiple parallel assays in the same data objects. This way we can keep all elements well organized together.

We should consider providing alias or renaming "standardize" to "scale" in mia but that's a separate issue (note to myself).

Reply:
Is it intentional here to do "scale" operation for relative abundances?

Usually one would do scaling for something more normal-like, such as log relative abundances or Aitchison family or transformations (clr, ilr, alr, rclr) etc. I am afraid that some modeling assumptions may be broken if we scale relative abundances directly.